external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -1198,20 +1196,14 @@ fn file_sources(
                     // there is a fair number of unwraps here but if we can't read the files
                     // that by all accounts should be in the file system, then there is not much
                     // we can do so it's better to fail so that we can investigate
-                    let mut vfs_file = overlay_fs
-                        .join(fname.as_str())
-                        .unwrap()
-                        .open_file()
-                        .unwrap();
+                    let vfs_file_path = overlay_fs.join(fname.as_str()).unwrap();
+                    let mut vfs_file = vfs_file_path.open_file().unwrap();
                     let _ = vfs_file.read_to_string(&mut contents);
                     let fhash = FileHash::new(&contents);
                     // write to top layer of the overlay file system so that the content
-                    // is immutable for the duration of compliation and symbolication
-                    let mut vfs_file = overlay_fs
-                        .join(fname.as_str())
-                        .unwrap()
-                        .create_file()
-                        .unwrap();
+                    // is immutable for the duration of compilation and symbolication
+                    let _ = vfs_file_path.parent().create_dir_all();
```

Comment:
With the added in-memory layer, the files do not yet exist in this layer and file creation fails if the path is not there, so we must create it explicitly.